enterprise/cmd/frontend/worker/auth/perms_syncer_cleaner.go

```diff
@@ -0,0 +1,110 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sourcegraph/sourcegraph/cmd/worker/job"
+	workerdb "github.com/sourcegraph/sourcegraph/cmd/worker/shared/init/db"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/env"
+	"github.com/sourcegraph/sourcegraph/internal/goroutine"
+	"github.com/sourcegraph/sourcegraph/internal/metrics"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+var _ job.Job = (*permissionsSyncerCleaner)(nil)
+
+// permissionsSyncerCleaner is a worker responsible for cleaning up processed
+// permission sync jobs.
+type permissionsSyncerCleaner struct{}
```

Comment:
Nitpick, but I'd go with this:

```suggestion
type permissionSyncJobCleaner struct{}
```